File: sample-ext1/src/logger.ts

type LogFunc = (msg: string) => void;

export class Logger {
    func: LogFunc;
    constructor(func: LogFunc){
        this.func = func;
    }

    info(msg: string){
        this.log("info", msg);
    }

    error(msg: string){
        this.log("error", msg);
    }

    private log(type: string, msg: string){
        const date = new Date();
        this.func(`${date.toLocaleString()}:${date.getMilliseconds()} [${type}] ${msg}`);
    }
}

File: vba-language-server/VBALanguageServer/Logger.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VBALanguageServer {
	static class Logger {
        public static void Info(string msg) {
            Write("Info", msg);
        }

		public static void Error(string msg) {
			Write("Error", msg);
		}

		private static void Write(string Level, string msg) {
            var date = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff");
            Console.Error.WriteLine($"[{date}][{Level}] {msg}");
		}
    }
}
